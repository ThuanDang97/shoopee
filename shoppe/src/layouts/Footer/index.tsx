import { Box, Checkbox, Flex, Text, useMediaQuery } from '@chakra-ui/react'

// Components
import Icon from '@components/Icon'
import NavBar from '@components/NavBar'

// Constants
import { BREAKPOINTS, NAV_LIST, SOCIAL_ICONS } from '@constants/index'

const Footer = () => {
  const [isMobile] = useMediaQuery(BREAKPOINTS.MEDIUM)

  const socialNetworkIcons = isMobile ? SOCIAL_ICONS.slice(1) : SOCIAL_ICONS

  return (
    <Box
      as="footer"
      pt="37px"
      pb={{ base: '16px', md: '108px' }}
      borderTopColor="lightSilver"
      maxWidth="1248px"
      margin={{ base: '0px 16px', lg: '0 auto' }}
      borderTop={{ base: '0px', md: '1px' }}
    >
      <Flex
        justifyContent="space-between"
        flexDirection={{ base: 'column-reverse', md: 'unset' }}
      >
        <NavBar
          justifyContent="flex-start"
          flexDirection={{ base: 'column', md: 'unset' }}
          paddingTop={{ base: '42px', md: '10px' }}
          navList={NAV_LIST}
        />
        <Checkbox
          defaultChecked
          color="dark"
          borderColor="secondary"
          pt="10px"
          size="sm"
          display={{ base: 'inline-flex', md: 'none' }}
        >
          I agree to the website’s terms and conditions
        </Checkbox>
        <Flex
          borderBottom="1px"
          borderBottomColor="dark"
          justifyContent="space-between"
          alignItems="center"
        >
          <Text
            width="100%"
            color="secondary"
            marginBottom="10px"
            fontSize="small"
            display="inline-block"
          >
            Give an email, get the newsletter.
          </Text>
          <Icon
            alt="Right arrow"
            ml={{ base: 'unset', md: '128px' }}
            srcIcon="/images/iconRight.webp"
            width="25px"
            height="9px"
          />
        </Flex>
      </Flex>
      <Flex
        mt="48px"
        justifyContent={{ base: 'flex-start', md: 'space-between' }}
        alignItems={{ base: 'unset', md: 'center' }}
        flexDirection={{ base: 'column-reverse', md: 'unset' }}
      >
        <Text
          paddingTop={{ base: '38px', md: '0px' }}
          fontSize={{ base: 'extraSmall', md: 'base' }}
        >
          <Text
            as="b"
            variant="primary"
            fontSize={{ base: 'extraSmall', md: 'base' }}
          >
            © 2021 Shelly.
          </Text>{' '}
          Terms of use
          <Text
            as="b"
            variant="primary"
            fontSize={{ base: 'extraSmall', md: 'base' }}
          >
            {' '}
            and
          </Text>{' '}
          privacy policy
        </Text>
        <Flex gap={{ base: '16px', md: '30px' }} alignItems="center">
          <Box
            display={{ base: 'flex', md: 'none' }}
            alignItems="center"
            gap="16px"
          >
            <Text fontSize="extraSmall" variant="primary">
              Follow us
            </Text>
            <Box
              as="span"
              borderBottom="1px"
              borderBottomColor="dark"
              width="47px"
            />
          </Box>
          {socialNetworkIcons.map((icon) => (
            <Icon
              width={{ base: '12px', md: '18px' }}
              height={{ base: '12px', md: '18px' }}
              key={icon.id}
              alt={icon.alt}
              srcIcon={icon.iconURL}
            />
          ))}
        </Flex>
      </Flex>
    </Box>
  )
}

export default Footer
