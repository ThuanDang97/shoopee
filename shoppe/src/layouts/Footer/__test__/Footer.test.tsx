// Utils
import { render } from '@utils/testUtils'

// Components
import Footer from '@layouts/Footer'

describe('Footer render', () => {
  // jest.mock('@chakra-ui/react', () => {
  //   // --> Original module
  //   const originalModule = jest.requireActual('@chakra-ui/react')

  //   return {
  //     __esModule: true,
  //     ...originalModule,
  //     useMediaQuery: jest.fn().mockImplementation(() => ({
  //       isMobile: true,
  //     })),
  //   }
  // })

  it('Should Footer match Snapshot', () => {
    const { container } = render(<Footer />)

    expect(container).toMatchSnapshot()
  })
})
