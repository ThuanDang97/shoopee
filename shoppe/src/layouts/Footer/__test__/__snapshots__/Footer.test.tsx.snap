// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render Should Footer match Snapshot 1`] = `
<div>
  <footer
    class="css-1frnbgd"
  >
    <div
      class="css-15g748d"
    >
      <nav
        class="css-5mlst9"
      >
        <ul
          class="css-js3rfl"
          role="list"
        >
          <li
            class="css-tnjslz"
          >
            <a
              href="/"
            >
              Contact
            </a>
          </li>
          <li
            class="css-tnjslz"
          >
            <a
              href="/"
            >
              Terms of services
            </a>
          </li>
          <li
            class="css-tnjslz"
          >
            <a
              href="/"
            >
              Shipping and Return
            </a>
          </li>
        </ul>
      </nav>
      <label
        class="chakra-checkbox css-a1ldsk"
        data-checked=""
      >
        <input
          checked=""
          class="chakra-checkbox__input"
          style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
          type="checkbox"
          value=""
        />
        <span
          aria-hidden="true"
          class="chakra-checkbox__control css-kn2htf"
          data-checked=""
        >
          <div
            class="css-0"
            style="display: flex; align-items: center; justify-content: center; height: 100%;"
          >
            <svg
              class="css-1v5j1c"
              style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16;"
              viewBox="0 0 12 10"
            >
              <polyline
                points="1.5 6 4.5 9 10.5 1"
              />
            </svg>
          </div>
        </span>
        <span
          class="chakra-checkbox__label css-1p6mdqm"
          data-checked=""
        >
          I agree to the website’s terms and conditions
        </span>
      </label>
      <div
        class="css-11xbhwq"
      >
        <p
          class="chakra-text css-17znp9"
        >
          Give an email, get the newsletter.
        </p>
        <figure
          class="css-16dh40w"
        >
          <img
            alt="Right arrow"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            src="/_next/image?url=%2Fimages%2FiconRight.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2FiconRight.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2FiconRight.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2FiconRight.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2FiconRight.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2FiconRight.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2FiconRight.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2FiconRight.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2FiconRight.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2FiconRight.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2FiconRight.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </figure>
      </div>
    </div>
    <div
      class="css-82b87w"
    >
      <p
        class="chakra-text css-11rc6k7"
      >
        <b
          class="chakra-text css-1qtkha6"
        >
          © 2021 Shelly.
        </b>
         
        Terms of use
        <b
          class="chakra-text css-1qtkha6"
        >
           
          and
        </b>
         
        privacy policy
      </p>
      <div
        class="css-13zimb6"
      >
        <div
          class="css-o5xvzg"
        >
          <p
            class="chakra-text css-1bz4vjt"
          >
            Follow us
          </p>
          <span
            class="css-13xadkr"
          />
        </div>
        <figure
          class="css-1dvkee8"
        >
          <img
            alt="LinkIn Icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            src="/_next/image?url=%2Fimages%2FinIcon.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2FinIcon.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2FinIcon.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2FinIcon.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2FinIcon.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2FinIcon.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2FinIcon.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2FinIcon.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2FinIcon.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2FinIcon.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2FinIcon.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </figure>
        <figure
          class="css-1dvkee8"
        >
          <img
            alt="Facebook icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            src="/_next/image?url=%2Fimages%2Ffacebook.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Ffacebook.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ffacebook.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </figure>
        <figure
          class="css-1dvkee8"
        >
          <img
            alt="Instagram icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            src="/_next/image?url=%2Fimages%2Finstagram.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Finstagram.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Finstagram.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Finstagram.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Finstagram.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Finstagram.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Finstagram.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Finstagram.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Finstagram.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Finstagram.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Finstagram.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </figure>
        <figure
          class="css-1dvkee8"
        >
          <img
            alt="Twitter icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            src="/_next/image?url=%2Fimages%2Ftwitter.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Ftwitter.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ftwitter.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </figure>
      </div>
    </div>
  </footer>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
